### PR TITLE
feat: add GitHub changelog notes generator

### LINKED ISSUE #1106: Add support for GitHub's automaticly generated release notes
It's possible to hit an API and generate what would be populated for GitHub's automatically generated release notes.

It would be nice to support this, so that folks can use release-please in conjuntion with the GitHub feature.

### PR DESCRIPTION
**Pull request by chingor13**
Fixes #1106 

Allows configuration in the manifest config as `changelog-type: 'github'` or via the CLI `--changelog-type=github`.

### CODE REVIEW

File: src/changelog-notes/github.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+export class GitHubChangelogNotes implements ChangelogNotes {
+  private github: GitHub;
+  constructor(github: GitHub) {
+    this.github = github;
+  }
+  async buildNotes(
+    _commits: ConventionalCommit[],
+    options: BuildNotesOptions
+  ): Promise<string> {
+    return await this.github.generateReleaseNotes(
```
Comment (bcoe) [inline src/changelog-notes/github.ts:28]: Neat, so this will just grab the release notes generated by the GitHub API?

File: src/factory.ts [hunk 2]
```diff
@@ -111,7 +118,15 @@ export async function buildStrategy(
     bumpMinorPreMajor: options.bumpMinorPreMajor,
     bumpPatchForMinorPreMajor: options.bumpPatchForMinorPreMajor,
   });
-  const strategyOptions: StrategyOptions = {
+  const changelogNotes = buildChangelogNotes({
+    type: options.changelogType || 'default',
+    github: options.github,
+    changelogSections: options.changelogSections,
+    // commitPartial: options.commitPartial,
```
Comment (bcoe) [inline src/factory.ts:125]: Do we need these commented out, can they just be deleted?

File: src/changelog-notes/github.ts (hunk 1 above)
Comment (chingor13) [reply src/changelog-notes/github.ts:28]: Yes I added the GitHub doc link to the associated issue.

File: src/factory.ts (hunk 2 above)
Comment (chingor13) [reply src/factory.ts:125]: We can probably delete. Those are options used by ruby to customize the changelog templates and we considering exposing them to the configuration options but did not.

File: src/factory.ts [hunk 3]
```diff
@@ -111,7 +118,15 @@ export async function buildStrategy(
...
+    // headerPartial: options.headerPartial,
+    // mainTemplate: options.mainTemplate,
```
Comment (bcoe) [inline src/factory.ts:127]:
```suggestion
```

**Review by bcoe — APPROVED** [review]

**Pull request merged**